File: Python_Scripting_for_Computational_Science_Third_Edition/efficiency/regex/regexread.py
#!/usr/bin/env python
import re, sys, math
from numpy import zeros
# find no of lines in the file (= no of items in the array)
f = open(sys.argv[1], 'r')
nlines = 0
for line in f:
    nlines += 1
f.close()
n = int(math.ceil(nlines**(1/3.0)))  # careful rounding!
print n
a = zeros((n,n,n))

# read file into a:
f = open(sys.argv[1], 'r')
for line in f:
    m = re.search(r'\[(\d+),(\d+),(\d+)\]=(.*)', line)
    if m:
        a[int(m.group(1)),int(m.group(2)),int(m.group(3))] = float(m.group(4))
f.close()
print "a[1,1,3]=",a[1,1,3]

File: Python_Scripting_for_Computational_Science_Third_Edition/efficiency/regex/regexread.pl
# find no of lines in the file (= no of items in the array)
open F, "<$ARGV[0]";
$nlines = 0;
while (<F>) { $nlines += 1; }
close F;

$n = int($nlines**(1/3.0) + 0.5);  # careful rounding!
$total_length = $nlines;
# use 1D array here
@a = (0.0)x$total_length;
print "$n\n";

open F, "<$ARGV[0]";
while (<F>) {
    if (/\[(\d+),(\d+),(\d+)\]=(.*)/) {
	$i = $1*$n*$n + $2*$n + $3;
	$a[$i] = $4;
	#print "[$1,$2,$3] -> $i = $4\n";
    }
}
close F;
print "a[3]=$a[3]\n";
